### ISSUE TITLE
Element Send Keys limited in usefulness

### ISSUE THREAD

**Issue by twalpole** [mozilla/geckodriver#245]
In a previous version of the webdriver spec, a NULL character in the keys sent to an element was interpreted as release the modifier keys, and allowed for things like selecting all the characters in a field and deleting them CTRL - A - NULL - BACKSPACE.  This behavior still works with chromedriver but geckodriver/marionette seem to consider the NULL a key to be sent to the element (as the current version of the webdriver spec would seem to imply) and leaves the ctrl modifier on when backspace is pressed too.   I'm not sure if this was an intentional change in the spec or an oversight when it was updated, but it does limit the usefulness of sending keys to an element.  Note - it can't be split into multiple send keys since each call resets the selection of characters in the field. -  See also: https://github.com/SeleniumHQ/selenium/issues/2704

**Comment by andreastt** [comment]
I don’t think the WebDriver specification implies anything related to this, as the _Element Send Keys_ command hasn’t been defined yet. If null is meant to release all modifiers, we can certainly specify that so that we don’t break backwards compatibility unnecessarily.

**Comment by andreastt** [comment]
Spec tracking bug for _Element Send Keys_: https://github.com/w3c/webdriver/issues/346

**Comment by twalpole** [comment]
@andreastt   Element Send Keys (/session/{session id}/element/{element id}/value) does appear to have been defined - https://w3c.github.io/webdriver/webdriver-spec.html#element-send-keys (you may have been thinking of the keyboard actions ?).  A previous version of the spec that mentioned the NULL behavior is https://www.w3.org/TR/2015/WD-webdriver-20150902/#sendkeys

**Comment by andreastt** [comment]
No, that’s what I was saying: _Element Send Keys_ has not been defined yet, but we ought to define it so we can have interoperability between browsers.

**Comment by p0deje** [comment]
With element send keys being defined now, it is expected to work the way it used to work on OSS dialect, which means that the following example should produce "Hello":

```ruby
require 'selenium-webdriver'

driver = Selenium::WebDriver.for(:firefox)
driver.get "http://google.com"
driver.find_element(css: '[name="q"]').send_keys(:shift, "h", :null, "ello")
```

Currently, it produces "HELLO", so proper implementation is pending.
